docs/source/development/poetry.rst

```diff
@@ -0,0 +1,108 @@
+.. _poetry-link:
+
+Poetry
+======
+
+We are using `Poetry <https://python-poetry.org/docs/>`_ for the dependency management in both ``sds-data-manager`` and ``imap_processing``. All dependency management should go through Poetry.
+
+.. _poetry-overview-link:
+
+Poetry overview
+---------------
+
+Poetry is a tool which provides the ability to manage :ref:`dependencies <poetry-dependency-link>` in Python. Although there are lots of tools which do this, including the built-in pip, Poetry has several specific advantages:
+
+#. Poetry allows you to "lock" dependency versions in place.
+#. Poetry will automatically determine if dependencies are compatible.
+#. Poetry can provide dependency groups, which allow you to specify which subset of dependencies you would like to install.
+
+These advantages ensure that we can all install the same versions of tools, that these versions are compatible with each other, and that we can manage dependencies in a fine tuned way. When you run ``poetry install`` to install your dependencies, you can be confident that they are the dependency versions that other people have been testing with.
+
+Poetry also provides tools for automatically bumping dependency versions only where it makes sense. You can have specify when a dependency should get a version change using Poetry's extensive `dependency specification <https://python-poetry.org/docs/dependency-specification/>`_ formatting.
+
+Finally, Poetry provides a :ref:`shell <poetry-shell-link>`, which you can use as a virtual environment. This shell is automatically tied to the project directory, and allows developers to install dependency versions in an isolated environment. This means that, for example, if you have a different project using the same package with a different version, you can have each project with it's own version kept seperate from conflicts.
+
+We have a :ref:`Poetry style guide<poetry-shell-link>` for specific recommendations about using Poetry in these projects.
```

Comment:
Is this supposed to link to [this](https://imap-processing.readthedocs.io/en/latest/development/style_guide/poetry-environment.html) instead?

Reply:
Yes, good catch